pandas/core/config_init.py

```diff
@@ -858,7 +858,7 @@ def register_converter_cb(key: str) -> None:
 with cf.config_prefix("future"):
     cf.register_option(
         "infer_string",
-        False,
+        True if os.environ.get("PANDAS_FUTURE_INFER_STRING", "0") == "1" else False,
```

Comment:
can we make this something that is explicitly FOR_TESTING_ONLY or something to clarify users shouldn't use it

Reply:
I would personally prefer to keep the name a bit more readable. We did the same for CoW, I don't think we had any problems with it? 
I also want to use this for testing in downstream packages, and would encourage others to do this as well (of course, that is still in a context of testing, but I feel that FOR_TESTING_ONLY would indicate it's only for testing for pandas devs)

(in general, there is not really a back compat issue with the env variable, in pandas 3.0 we will just start ignoring it, which is even easier than the explicit `pd.options` option, which will start raising an error at some point)